Fabric.Terminology.SqlServer/Persistence/SqlClientTermValueSetRepository.UoWOperation.cs

```diff
@@ -0,0 +1,138 @@
+﻿namespace Fabric.Terminology.SqlServer.Persistence
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using CallMeMaybe;
+
+    using Fabric.Terminology.Domain;
+    using Fabric.Terminology.Domain.Models;
+    using Fabric.Terminology.Domain.Persistence;
+    using Fabric.Terminology.SqlServer.Models.Dto;
+    using Fabric.Terminology.SqlServer.Persistence.UnitOfWork;
+
+    internal partial class SqlClientTermValueSetRepository
+    {
+        private static ValueSetCodeDto ConvertToValueSetCodeDto(Guid valueSetGuid, ICodeSystemCode code) =>
+            new ValueSetCodeDto(code) { ValueSetGUID = valueSetGuid };
+
+        private AddRemoveCodeOperations PrepareAddRemoveCodes(
+            Guid valueSetGuid,
+            IReadOnlyCollection<ICodeSystemCode> codesToAdd,
+            IReadOnlyCollection<ICodeSystemCode> codesToRemove)
+        {
+            var currentCodeDtos = this.uowManager.GetCodeDtos(valueSetGuid);
+
+            //// There could be duplicates between the codesToAdd and codesToRemove collections.
```

Comment:
Why could there be duplicates? I'd prefer for the API to fail fast because the intent would be ambiguous: there's no way for you to know whether they meant "add this, then remove it" or "remove this, then add it." The caller/client should shoulder the responsibility of keeping track of those details.